R/create_protocol.R

```diff
@@ -104,6 +111,8 @@ create_protocol <- function(
   assert_that(is.string(short_title), nchar(short_title) <= 20)
   assert_that(is.date(as.Date(date)))
   assert_that(is.character(authors))
+  assert_that(is.character(orcids),
+              length(authors) == length(orcids))
```

Comment:
Thanks for pointing this out. I checked and it is possible to set a value to NA and `ymlthis::yml_author()` will correctly deal with this:

``` r
library(ymlthis)
yml_empty() %>%
  yml_author(name = c("a", "b"), orcid = c(NA, "d"))
#> ---
#> author:
#> - name: a
#> - name: b
#>   orcid: d
#> ---

length(c("a", "b")) == length(c(NA, "d"))
#> [1] TRUE
```

<sup>Created on 2021-06-02 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

I will document this so that users are informed what to do when one of the authors does not have an ORCID.